ruby_event_store/lib/ruby_event_store/immediate_async_dispatcher.rb

```diff
@@ -0,0 +1,15 @@
+module RubyEventStore
+  class ImmediateAsyncDispatcher
+    def initialize(scheduler:)
+      @scheduler = scheduler
+    end
+
+    def call(subscriber, _, serialized_event)
+      @scheduler.call(subscriber, serialized_event)
+    end
```

Comment:
Notice that if scheduler `call` API was identical to dispatcher's API, you could simply substitute scheduler for a dispatcher without the need for `ImmediateAsyncDispatcher`. Wouldn't that be nice?